### DISCUSSION TITLE
Bad Request: not enough rights to send text messages to the chat

### DISCUSSION THREAD

**Discussion by bitdom8** [telegraf/telegraf#1833] [Q&A] [answered]
Don't know why telegraf check my local running server along with server one, but doesn't check if a group admin adds bot to his group doesn't check if `not enough rights to send text messages to the chat`.

Telegraf shuts down the server and linux restart the telegram nodejs to finish mess `ExecStart=npm run preview2`

**Comment by MKRhere** [comment]
I don't understand the problem. Can you describe better step by step?

**Comment by bitdom8** [comment]
I added the bot to a channel but not make him admin. Then this errors pop out and stop the node server @MKRhere

**Reply by MKRhere** [reply]
You cannot add a bot to a channel without giving admin. It will be removed. You can only add the bot to groups without admin.

**Comment by M1Ergash** [comment]
> Don't know why telegraf check my local running server along with server one, but doesn't check if a group admin adds bot to his group doesn't check if `not enough rights to send text messages to the chat`.
> 
> Telegraf shuts down the server and linux restart the telegram nodejs to finish mess `ExecStart=npm run preview2`

I have same issue, error is "Bad Request: not enough rights to send text messages to the chat". I tried to change my bot in the channel.

**Reply by divinets** [reply]
Is it worked?

**Reply by M1Ergash** [reply]
Unfortunately no, I'm still googling, but it looks like this is a new problem that no one has encountered yet. do you have any idea how to solve it?

**Reply by divinets** [reply]
It was happened 4 hours ago. I think this is a new policy of Telegram maybe or just an error on Telegram API.

**Reply by divinets** [reply]
I have no idea how to fix it.

**Reply by maximq** [reply]
> Unfortunately no, I'm still googling, but it looks like this is a new problem that no one has encountered yet. do you have any idea how to solve it?

I also faced this problem
Does this seem like a widespread problem?
My bot worked non-stop for a year or two

**Reply by divinets** [reply]
This error is only for Channels not for Groups.

**Reply by MKRhere** [reply]
Telegram Bot API is having issues since a few hours ago. If your bot suddenly started throwing errors, be patient, it should be fixed when they figure out the update.

**Reply by 94wogus-quantit** [reply]
same issue here

**Reply by M1Ergash** [reply]
yes, exactly, I’m interested in the fact that my bot worked without a break for a couple of months

**Reply by DmitriyLyalyuev** [reply]
The same here. :( Any advice?

**Reply by Custler** [reply]
I have the same issue
`{"ok":false,"error_code":400,"description":"Bad Request: not enough rights to send text messages to the chat"}`

**Reply by dmitry-shanko** [reply]
Same, posted an issue to support:(
My bot stopped sending messages since ~01:30 UTC (01:25 last message)

**Reply by M1Ergash** [reply]
I checked my bot in a group, It is working...
But I need it to be worked in the channels

**Reply by divinets** [reply]
01:29 UTC last message for me

**Reply by sashok2k** [reply]
Same here

**Reply by vsile** [reply]
The same issue for @RemindMegaBot

**Reply by hypernova7** [reply]
> Attention ⚠: Seems like Telegram Bot API is having trouble, potentially due to an upcoming update, and responds with a "not enough rights" error to many API calls.
>
> Be assured this is temporary. Your bots should be back in operation once they fix the issue.
>
> Posted on: https://t.me/TelegrafJSChat/124313

**Reply by DmitriyLyalyuev** [reply]
As I understand it's not related to Telegraf directly. I'm using python libs too and it's have the same issue. So it's on Telegram side. Waiting for fix.

**Reply by LingamYoni** [reply]
So it's on Telegram side - YES!

**Reply by EnkhAmar** [reply]
same issue here. Please let me know when it is fixed

**Reply by Choco-milk-for-u** [reply]
i thought it was my problem

**Reply by IkromKhakimov** [reply]
Same issue

**Reply by souz-odin** [reply]
Same issue

**Reply by Fleckinger** [reply]
Same problem with java library.

**Reply by mb-mal** [reply]
same problem even through Postman direct request. So could be not an telegraf issue.

**Reply by nm-dk** [reply]
Same issue on all channels (groups are ok). Bad morning news...

**Reply by ghost** [reply]
same issue
how to fix ?

**Reply by felipebcv** [reply]
Here's the same issue, the strange thing is that I have bots in groups and they are working, only the bot in the channel has stopped.

**Reply by hpmpm** [reply]
yep same here not working with channel.

**Reply by kiukishenkaec** [reply]
Да, на прямую общение с ботом происходит, а вот в каналы сообщения он разсылать не может
![image](https://github.com/telegraf/telegraf/assets/22501630/b5019428-3f41-409e-8aaa-b960d857ed27)

**Reply by yashkakkar** [reply]
All Bot Stopped.. looks like alien 👽 Invasion!!

**Reply by osirisads** [reply]
Oh MAN .. same here, Group works fine but not Channel. It's all dead!

**Reply by IlluvatarEru** [reply]
@MKRhere as you can see, there must have been a recent update that introduced a new bug. It is impacting a lot of bots worldwide, can you either roll back or release a patch to fix?

**Reply by MKRhere** [reply]
@IlluvatarEru It's not a Telegraf update that caused this, but Telegram's Bot API itself. We can only wait until it's resolved.

**Reply by vahidsabet** [reply]
It's still not working but groups

**Comment by MKRhere** [comment, marked as answer]
Attention ⚠: Seems like Telegram Bot API is having trouble, potentially due to an upcoming update, and responds with a "not enough rights" error to many API calls.

Be assured this is temporary. Your bots should be back in operation once they fix the issue.

Posted on: [t.me/TelegrafJSChat/124313](https://t.me/TelegrafJSChat/124313)

I'm locking this conversation to stop the flood of spam. Yes, it seems to affect channels, but not groups. Yes, your bot was working fine before. It's not your fault, and we cannot fix it. The only possible solution right now is to run an older version of [Telegram Bot API](https://github.com/tdlib/telegram-bot-api) yourself.

Edit: Seems to be resolved now, your bots should be working normally again.
